packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -240,6 +240,34 @@ export default declare((api, opts) => {
     }
   }
 
+  function hasMoreThanOneBinding(node) {
+    // ArrayPattern
+    if (node.elements) {
```

Comment:
Rather than duck-checking the type of `node`, can we do something like this?

```js
if (t.isArrayPattern(node) {
} else if (t.isObjectPattern(node)) {
...
```

Reply:
"duck-checking" is inspired by "duck-typing", but probably it's not the correct term :joy:

Reply:
Sure I'll add that now

Reply:
Updated! I didn't realize these interface methods existed - definitely safer checks and will likely help down the line for a TS conversion.